File: C/Communication entre processus/programme_pere44.c
#include <stdio.h>
#include <unistd.h>
#include <stdlib.h>

int main() {
    int fd[2];
    pipe(fd);

    pid_t pid = fork();

    if(pid == 0) 
    {
        // convertir fd[0] en chaîne
        char r[10];
        sprintf(r, "%d", fd[0]);

        // mettre dans variable d'environnement
        setenv("FD", r, 1);

        // lancer le programme fils
        execl("./fils", "fils", NULL);
    }
    else
    {
        char msg[] = "Hellooooo depuis le père";

        write(fd[1], msg, sizeof(msg));
    }

    return 0;
}

File: Python/Communication entre processus/programme_pere44.py
import os

r, w = os.pipe()

pid = os.fork()

if pid == 0:
    # mettre le fd dans variable d'environnement
    os.environ["FD"] = str(r)

    # lancer le programme fils
    os.execv("/usr/bin/python3", ["python3", "programme_fils44.py"])

else:
    msg = "Hello depuis le père"
    os.write(w, msg.encode())